VCVio/OracleComp/SimSemantics/Simulate.lean

```diff
@@ -20,17 +20,28 @@ For example `idOracle` has no effect upon simulation, and we should apply that f
 
 open OracleSpec Prod
 
-universe u v w
+universe u u' v v' w w' z
 
-variable {ι : Type u} {ιₜ : Type v} {spec : OracleSpec ι} {specₜ : OracleSpec ιₜ}
-  {α β σ : Type w}
+variable {ι ιₜ : Type*} {spec : OracleSpec ι} {specₜ : OracleSpec ιₜ}
+  {α β σ : Type u}
+
```

Comment:
Version of `SimOracle` that allows an arbitrary monad transformer `T` to be applied to the output state